numpy/lib/shape_base.py

```diff
@@ -354,14 +354,14 @@ def dstack(tup):
     """
     Stack arrays in sequence depth wise (along third axis).
 
-    Takes a sequence of arrays and stack them along the third axis
-    to make a single array. Rebuilds arrays divided by `dsplit`.
-    This is a simple way to stack 2D arrays (images) into a single
-    3D array for processing.
+    This is equivalent to concatenation along the third dimension, after 2d
+    arrays of shape `(M,N)` have been reshaped to `(M,N,1)` and 1d arrays of
+    shape `(N,)` have been reshaped to `(1,N,1)`. Rebuilds arrays divided by
+    `dsplit`.
 
-    This function continues to be supported for backward compatibility, but
-    you should prefer ``np.concatenate`` or ``np.stack``. The ``np.stack``
-    function was added in NumPy 1.10.
+    This function is meant for use on arrays with 3 dimensions or less. The
```

Comment:
Not sure on the 3-D or less description here.  It might not seem sensible, but there is no restriction. Maybe "makes most sense".